include/pybind11/numpy.h

```diff
@@ -478,7 +480,7 @@ class array : public buffer {
         }
 
         auto tmp = reinterpret_steal<object>(api.PyArray_NewFromDescr_(
-            api.PyArray_Type_, descr.release().ptr(), (int) ndim, shape.data(), strides.data(),
+            api.PyArray_Type_, descr.release().ptr(), (int) ndim, shape.v.data(), strides.v.data(),
```

Comment:
Btw you only use the lvalue (const) ref in passing it to `default_strides()` though? So maybe if you remove the mutable `&` overload and leave just `&&` and `const &` there would be no ambiguity?